modules/ptp-configuring-linuxptp-services-as-ha-bc-for-dual-nic.adoc

```diff
@@ -0,0 +1,150 @@
+
+You can configure `linuxptp` services (`ptp4l`, `phc2sys`) as a highly available boundary clock (T-BC) for dual Intel e810 Salem channel dual NIC hardware.
+You connect each NIC to the same upstream leader clock with separate `ptp4l` instances for each NIC feeding the downstream clocks.
+
+Create two `PtpConfig` custom resource (CR) objects for each NIC and a third `PtpConfig` CR that configures high availability between the two NICs.
+
+The third `ptpConfig` CR configures the `ptp4l` and `phc2sys` services for network clock synchronization.
+The third CR adds profiles for the `ptp4l` configurations under the `spec.profile.ptpSettings.haProfiles` key and passes the kernel socket path of those profiles to the `phc2sys` service.
+When a `ptp4l` failure occurs, the `phc2sys` service switches to the  backup `ptp4l` configuration.
+
+.Prerequisites
+
+* Install the OpenShift CLI (`oc`).
+
+* Log in as a user with `cluster-admin` privileges.
+
+* Install the PTP Operator.
+
+* Configure a cluster node with Intel e810 Salem channel dual NIC.
+
+.Procedure
+
+. Create two separate `PtpConfig` CRs, one for each NIC, using the reference CR in "Configuring linuxptp services as a boundary clock" as the basis for each CR.
+
+.. Create `ha-boundary-clock-ptp-config-nic1.yaml`, specifying an empty string for `phc2sysOpts`.
+For example:
++
+[source,yaml]
+----
+apiVersion: ptp.openshift.io/v1
+kind: PtpConfig
+metadata:
+  name: ha-boundary-clock-ptp-config-nic1
+  namespace: openshift-ptp
+spec:
+  profile:
+  - name: "profile1"
+    ptp4lOpts: "-2 --summary_interval -4"
+    ptp4lConf: | <1>
+      [ens5f1]
+      masterOnly 1
+      [ens5f0]
+      masterOnly 0
+    [...]
+    phc2sysOpts: "" <2>
+----
+<1> Specify the required interfaces to start `ptp4l` as a boundary clock. For example, `ens5f0` synchronizes from a grandmaster clock and `ens5f1` synchronizes connected devices.
+<2> Set `phc2sysOpts` with an empty string.
+These values are populated from the `spec.profile.ptpSettings.haProfiles` field of the `PtpConfig` CR that configures high availability.
+
+.. Create `ha-boundary-clock-ptp-config-nic2.yaml`, specifying an empty string for `phc2sysOpts`.
+For example:
++
+[source,yaml]
+----
+apiVersion: ptp.openshift.io/v1
+kind: PtpConfig
+metadata:
+  name: ha-boundary-clock-ptp-config-nic2
+  namespace: openshift-ptp
+spec:
+  profile:
+  - name: "profile2"
+    ptp4lOpts: "-2 --summary_interval -4"
+    ptp4lConf: |
+      [ens7f1]
+      masterOnly 1
+      [ens7f0]
+      masterOnly 0
+    [...]
+    phc2sysOpts: ""
+----
+
+. Create the `PtpConfig` CRs that configures PTP for both NICs:
++
+[source,terminal]
+----
+$ oc create -f ha-boundary-clock-ptp-config-nic1.yaml && \
+oc create -f ha-boundary-clock-ptp-config-nic2.yaml
+----
+
+. Create the `PtpConfig` CR that configures T-BC high availability
+For example:
+
+.. Create `ptp-config-for-ha.yaml`:
++
+[source,yaml]
+----
+include::snippets/ztp_PtpConfigForHA.yaml[]
```

Comment:
missing snippet?

Reply:
It's a href to the file. It resolves at build time.